utils.js

```diff
@@ -29,12 +32,17 @@ export const has = (prop, obj) =>
   Object.prototype.hasOwnProperty.call(obj, prop);
 
 export const isEmpty = (obj) => {
-  for (var key in obj) {
-    if (obj.hasOwnProperty(key)) {
-      return false;
-    }
+  if (isNil(obj)) {
+    return false;
+  } else if (isArray(obj)) {
+    return obj.length === 0;
+  } else if (isString(obj)) {
+    return obj === '';
+  } else if (isObject(obj)) {
+    return Object.keys(obj).length === 0 && obj.constructor === Object;
```

Comment:
It's needed because there is a possibility of non-empty object being detected as an empty object, for example:

```
Object.keys(new Date()).length === 0; // true
Object.keys(new Number()).length === 0; // true
```